axelor-account/src/main/java/com/axelor/apps/account/service/invoice/generator/tax/TaxInvoiceLine.java

```diff
@@ -28,28 +31,37 @@
 import java.util.Map;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import javax.inject.Inject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class TaxInvoiceLine extends TaxGenerator {
 
   private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
-  public TaxInvoiceLine(Invoice invoice, List<InvoiceLine> invoiceLines) {
+  protected TaxAccountToolService taxAccountToolService;
+
+  @Inject
+  public TaxInvoiceLine(
+      Invoice invoice,
+      List<InvoiceLine> invoiceLines,
+      TaxAccountToolService taxAccountToolService) {
 
     super(invoice, invoiceLines);
+    this.taxAccountToolService = taxAccountToolService;
   }
 
   /**
    * Créer les lignes de TVA de la facure. La création des lignes de TVA se basent sur les lignes de
    * factures
    *
    * @return La liste des lignes de TVA de la facture.
+   * @throws AxelorException
    */
   @Override
-  public List<InvoiceLineTax> creates() {
+  public List<InvoiceLineTax> creates() throws AxelorException {
 
-    Map<TaxLine, InvoiceLineTax> map = new HashMap<>();
+    Map<List<Object>, InvoiceLineTax> map = new HashMap<>();
```

Comment:
Please do not use a list of object, instead use a collection that states explicitly what will be used as keys. You can create a nested class that contains a taxLine, or use a map of a map, like `Map<TaxLine, Map<Integer, InvoiceLineTax>>`